PiWeb.Api.Training.Events/Program.cs

```diff
@@ -0,0 +1,137 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR.Client;
+using PiWeb.Api.Training.Events.Events;
+
+namespace PiWeb.Api.Training.Events;
+
+public static class Program
+{
+    public static async Task Main()
+    {
+        Console.WriteLine("PiWeb Server Events - Training Client");
+        Console.WriteLine("=====================================");
+
+        Console.WriteLine();
+        ConsoleUI.WriteInfo("Find PiWeb Server (1/4)");
+
+        // Get the Server URL from user input, e.g. copied from the PiWeb Server UI "Autodiscovery address"
+        if (!TryGetServerUrl(out var serverUrl))
+            return;
+
+        Console.WriteLine();
+        ConsoleUI.WriteInfo($"Contacting {serverUrl} (2/4)");
+
+        // Check which authentication modes are available
+        var authenticationMode = await Authentication.GetAuthenticationMode(serverUrl);
+
+        Console.WriteLine();
+        ConsoleUI.WriteInfo("Creating event connection (3/4)");
+
+        // Try to connect to the /events endpoint
+        if (!TryCreateConnection(serverUrl, authenticationMode, out var connection))
+            return;
+
+        // Subscribe to the events you want to be informed about
+        SubscribeEvents(connection);
+
+        if (!await TryConnect(connection))
+            return;
+
+        Console.WriteLine();
+        ConsoleUI.WriteInfo("Listening for events... (READY)");
+
+        // Now simply wait for any incoming events
+        await Task.Delay(Timeout.Infinite);
+    }
+
+    private static bool TryGetServerUrl([NotNullWhen(true)] out Uri? serverUrl)
+    {
+        var serverUrlString = ConsoleUI.InputText("Server URL with port");
+
+        try
+        {
+            serverUrl = new Uri(serverUrlString);
+            return true;
+        }
+        catch (Exception exception)
+        {
+            ConsoleUI.WriteError(exception.ToString());
+            serverUrl = null;
+            return false;
+        }
+    }
+
+
```

Comment:
Too many empty lines. Also after the method.

Reply:
Done